ArduCopter/mode_smart_rtl.cpp

```diff
@@ -11,6 +11,13 @@
 
 bool ModeSmartRTL::init(bool ignore_checks)
 {
+    // Add check to prevent SmartRTL activation when disarmed
+    if (!motors->armed()) {
```

Comment:
In copter we always allow the user to switch to the mode if the vehicle is disarmed.  We do not allow users to arm in SmartRTL though so users are forced to be in a different mode when it comes time to arm.